scripts/app.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+// declare modules
+angular.module('Authentication', []);
+angular.module('Home', []);
+
+angular.module('BasicHttpAuthExample', [
+    'Authentication',
+    'Home',
+    'ngRoute',
+    'ngCookies'
+])
+ 
+.config(['$routeProvider', function ($routeProvider) {
+
+    $routeProvider
```

Comment:
@anijat-cuelogic use state provider rather than root provider, it will give you flexibility to share data between controllers.